atoma-proxy/src/server/middleware.rs

```diff
@@ -716,16 +757,15 @@ pub async fn handle_locked_stack_middleware(
                 1,
                 &[KeyValue::new("model", request_metadata.model_name.clone())],
             );
-            UNAVAILABLE_STACK_COUNTER_PER_USER.add(
-                1,
-                &[
-                    KeyValue::new(
-                        "stack_small_id",
-                        request_metadata.selected_stack_small_id.to_string(),
-                    ),
-                    KeyValue::new("user_id", request_metadata.user_id),
-                ],
-            );
+            if let Some(stack_small_id) = request_metadata.selected_stack_small_id {
+                UNAVAILABLE_STACK_COUNTER_PER_USER.add(
```

Comment:
Should we have similar prometheus metrics for the fiat payments ?

Reply:
Since fiat is post payment (it's always accepted by the node) you will never receive TOO_EARLY (this is when the stack was not found on the node. In case of fiat this will never happen, either you are in the white list or not.